contracts/Whitelist.sol

```diff
@@ -9,24 +10,51 @@ import "./packages/oz/Ownable.sol";
  * @notice The whitelist module keeps track of all valid Otoken contracts.
  */
 contract Whitelist is Ownable {
-    ///@notice mapping to track whitelisted product
+    ///@notice AddressBook module address
+    address public addressBook;
+
+    ///@dev mapping to track whitelisted products
     mapping(bytes32 => bool) internal whitelistedProduct;
+    ///@dev mapping to track whitelisted otokens
+    mapping(address => bool) internal whitelistedOtoken;
+
+    /**
+     * @dev constructor
+     * @param _addressBook AddressBook module address
+     */
+    constructor(address _addressBook) public {
+        require(_addressBook != address(0), "Invalid address book");
 
-    ///@notice emitted when owner whitelist a product
+        addressBook = _addressBook;
+    }
+
+    ///@notice emitted when the owner whitelists a new product
     event ProductWhitelisted(
         bytes32 productHash,
         address indexed underlying,
         address indexed strike,
         address indexed collateral
     );
 
+    ///@notice emitted when Otoken Factory module whitelist an otoken
+    event OtokenWhitelisted(address otoken);
+
     /**
-     * @notice check if a product is supported
+     * @notice check if sender is Otoken Factory module
+     */
+    modifier onlyFactory() {
+        require(msg.sender == AddressBookInterface(addressBook).getOtokenFactory(), "Sender is not Otoken Factory");
```

Comment:
I suggest adding the contract name to the require message, ( `Sender is not Otoken Factory` => `WhiteList: Sender is not Otoken Factory`, `WL:Sender is not Otoken Factory`)  that way it's much easier for us to know which contract is throwing errors while testing.